pkg/cmdutil/usage.go

```diff
@@ -0,0 +1,173 @@
+package cmdutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/algolia/cli/pkg/iostreams"
+	"github.com/algolia/cli/pkg/text"
+	"github.com/algolia/cli/pkg/utils"
+)
+
+type UsageEntry struct {
+	Title string
+	Body  string
+}
+
+func UsageFunc(IOStreams *iostreams.IOStreams, command *cobra.Command, flagUsages string) error {
+	entries := []UsageEntry{}
```

Comment:
IMHO, we should have a struct declared `[]UsageEntry{}` and we will add the methods `AddBasicUsage`, `AddFlags` etc on it.

Also, all these methods do not need to be public (capitalised).